crates/bevy_asset/src/asset_changed.rs

```diff
@@ -0,0 +1,458 @@
+//! Define the [`AssetChanged`] query filter.
+//!
+//! Like [`Changed`](bevy_ecs::prelude::Changed), but for [`Asset`]s.
+
+use crate::{AsAssetId, Asset, AssetId};
+use bevy_ecs::component::Components;
+use bevy_ecs::{
+    archetype::Archetype,
+    component::{ComponentId, Tick},
+    prelude::{Entity, Resource, World},
+    query::{FilteredAccess, QueryFilter, QueryItem, ReadFetch, WorldQuery},
+    storage::{Table, TableRow},
+    world::unsafe_world_cell::UnsafeWorldCell,
+};
+use bevy_utils::tracing::error;
+use bevy_utils::HashMap;
+use core::marker::PhantomData;
+use disqualified::ShortName;
+
+#[derive(Resource)]
+pub(crate) struct AssetChanges<A: Asset> {
```

Comment:
Basic doc strings for internal types please, especially linking the various parts of the machinery together. It's much easier to review and maintain!

Reply:
Added docs also noting the safety implications of this resource.